legacy/src/main/java/org/opensearch/sql/legacy/query/OpenSearchActionFactory.java

```diff
@@ -191,12 +201,7 @@ private static boolean isJoin(SQLQueryExpr sqlExpr, String sql) {
   }
 
   @VisibleForTesting
-  public static boolean shouldMigrateToQueryPlan(SQLQueryExpr expr, Format format) {
-    // The JSON format will return the OpenSearch aggregation result, which is not supported by the
-    // QueryPlanner.
-    if (format == Format.JSON) {
-      return false;
-    }
```

Comment:
Now there is no JSON format, I'm thinking should we ignore/fix the failed IT instead of adding a flag `bypassMigrateToQueryPlan`?

Reply:
main problem trying to solve with this flag is `CheckScriptContents.getScriptFieldFromQuery` needs access to a `SearchRequestBuilder` in `QueryAction`, which I think `QueryPlanQueryAction` doesn't expose. Will double check if there's another way

Reply:
I think for the tests that used this method, it's a simple fix to change the query to not have any aggregation. Doing so will avoid migrating to QueryPlanner as well.

Those are UTs testing the string operators, and the GROUP BY and ORDER BY aren't used anyway, so it should be safe to remove them.